packages/compass-schema/src/stores/schema-analysis-reducer.ts

```diff
@@ -21,72 +23,80 @@ import type { Circle, Layer, LayerGroup, Polygon } from 'leaflet';
 import { mongoLogId } from '@mongodb-js/compass-logging/provider';
 import type { SchemaThunkAction } from './store';
 import { UUID } from 'bson';
+import { isAction } from '../utils';
 
 const DEFAULT_SAMPLE_SIZE = 1000;
 
 const ERROR_CODE_MAX_TIME_MS_EXPIRED = 50;
 
-export function isAction<A extends AnyAction>(
-  action: AnyAction,
-  type: A['type']
-): action is A {
-  return action.type === type;
-}
-
-export type SchemaState = {
+export type SchemaAnalysisState = {
   analysisState: AnalysisState;
   errorMessage: string;
   schema: Schema | null;
+  schemaAccessor: SchemaAccessor | null;
```

Comment:
This is not state either, this is really way closer to service that should be passed through thunk argument. I linked one example for this in another place, for this one I guess it would be closer to [PipelineBuilder in compass-aggregations plugin](https://github.com/mongodb-js/compass/blob/40056b23274fc374e29864e2b9f25cbf77277920/packages/compass-aggregations/src/stores/store.ts#L126), it's just a service thats' relevant only for this plugin and we want to use in various actions